test/FastTests/Issues/RavenDB_13869.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using FastTests.Server.Basic.Entities;
+using Raven.Client.Documents.Session;
+using Raven.Tests.Core.Utils.Entities;
+using Xunit;
+
+namespace FastTests.Issues
+{
+    public class RavenDB_13869:RavenTestBase
+    {
+        
+        [Fact]
+        public void MissingFieldsDataShouldBeCleared()
+        {
+            using (var store = GetDocumentStore())
+            {
+                using (var session = store.OpenSession())
+                {
+                    //person
+                    session.Store(new User
+                    {
+                        Name = "John",
+                        LastName = "Doh"
+                    },"users/1");
+                    session.SaveChanges();
+                }
+
+                using (var session = store.OpenSession())
+                {
+                    var person = session.Load<Person>("users/1");
+                    var sessionOperations = (session as InMemoryDocumentSessionOperations);                    
+                    session.Advanced.Clear();
+
+                    session.Store(person, "users/2");                    
+                    session.SaveChanges();                    
+                }
+
+                using (var session = store.OpenSession())
+                {
+                    var person = session.Load<User>("users/2");
+                    Assert.Null(person.LastName);
```

Comment:
No sue if I follow - why is it `null`?

Reply:
It should be null, because we called session.clear, so we didn't send the not materialized fields back to the server

Reply:
It's a not direct method to test the cleanup behavior..

Reply:
but we called `session.Store(person, "users/2");` and `session.Clear` clears a state of UoW, not properties of entities - am I missing something?

Reply:
`User` vs `Person`, I think.
`Person` doesn't have `LastName` ?

Reply:
That's right @ayende , so after clear, we should not send this value to the server

Reply:
Can you use types that are defined in the test? It is _very_ hard to figure out inline

Reply:
Yeah, I haven't noticed this - `User` vs `Person`